lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -288,6 +288,24 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		case r.LocalAmt == 0:
 			break
 
+		// In case this request uses funding up to some maximum amount,
+		// we will call the specialized method for that.
+		case r.FundUpToMaxAmt != 0:
+			dustLimit := w.cfg.DustLimit
+			selectedCoins, localContributionAmt, changeAmt, err = CoinSelectUpToAmount(
+				r.FeeRate, r.FundUpToMaxAmt, dustLimit, coins,
+			)
+			if err != nil {
+				return err
+			}
+
+			// In case the selected amount is lower than the amount defined
+			// as the minimum with LocalAmt, we must return an error.
+			if localContributionAmt < r.LocalAmt {
```

Comment:
Is this `LocalAmt` set by the user's value of `local_funding_amount` in the RPC? If yes, then the comment on the `fund_max` in the `lightning.proto` is invalid, if the `local_funding_amount` acts as a minimum and the `fund_max` as a maximum.

Reply:
Ah, I see that the local amount is set to a minimum amount in the RPC server and is not the user's value. Maybe worth a sentence here.

Reply:
Yes, the `LocalAmt` is set to at least the minimum channel size here: https://github.com/lightningnetwork/lnd/pull/4029/commits/e9a0c14789fcc221447b5d23eb59a21ac13b322b#diff-674c79ba02b6e9c45a994388392689814f3538800351b2fd43fb3ade21e1effbR1892.

I've added a comment for clarification.

Reply:
> Ah, I see that the local amount is set to a minimum amount in the RPC server and is not the user's value. Maybe worth a sentence here.

@guggero  during testing I realized that using the `minchansize` from our config here puts a minimum size on our outgoing channel where as per definition this config parameter is supposed to only affect incoming channel sizes. Say I only allowed peers to open 5M sat channels to me I wouldn't be able to use `--fundmax` unless my wallet balance was >= 5M sats. 
Instead we could use the [`MinChanDefaultSize`](https://github.com/lightningnetwork/lnd/blob/670bd584986209b1a567cac36084b8e560c6c595/funding/manager.go#L77) here which is 20k sats.

Reply:
ooof is it perhaps worth adding separate "MinAmt" and "MaxAmt" to the request instead of repurposing the LocalAmt field?

Reply:
Yes I think that's cleaner. I've added a `FundMinAmt` field and adjusted the sanity checks.